packages/compass-workspace/src/components/tab.tsx

```diff
@@ -0,0 +1,326 @@
+  overflow: 'hidden',
+  fontWeight: 'bold',
+  fontSize: compassFontSizes.smallFontSize,
+  gridArea: 'tabName',
+});
+
+const tabTitleDarkThemeStyles = css({
+  color: uiColors.gray.base,
+});
+
+const tabTitleLightThemeStyles = css({
+  color: uiColors.gray.base,
+});
+
+const tabTitleSelectedDarkThemeStyles = css({
+  color: uiColors.green.light2,
+});
+
+const tabTitleSelectedLightThemeStyles = css({
+  color: uiColors.green.dark2,
+});
+
+const tabTitleFocusedStyles = css({
+  color: uiColors.focus,
+});
+
+const tabSubtitleStyles = css({
+  whiteSpace: 'nowrap',
+  textOverflow: 'ellipsis',
+  overflow: 'hidden',
+  fontSize: compassFontSizes.smallFontSize,
+  lineHeight: `${spacing[3]}px`,
+  gridArea: 'namespace',
+});
+
+const tabSubtitleLightThemeStyles = css({
+  color: uiColors.gray.base,
+});
+
+const tabSubtitleDarkThemeStyles = css({
+  color: uiColors.gray.light1,
+});
+
+const tabSubtitleSelectedLightThemeStyles = css({
+  color: uiColors.gray.dark1,
+});
+
+const tabSubtitleSelectedDarkThemeStyles = css({
+  color: uiColors.gray.light2,
+});
+
+type TabProps = {
+  title: string;
+  darkMode?: boolean;
+  isSelected: boolean;
+  onSelect: () => void;
+  onClose: () => void;
+  renderIcon: (
+    iconProps: Partial<React.ComponentProps<typeof Icon>>
+  ) => JSX.Element;
+  tabContentId: string;
+  subtitle: string;
+};
+
+function UnthemedTab({
+  darkMode,
+  title,
+  isSelected,
+  onSelect,
+  onClose,
+  tabContentId,
+  renderIcon,
+  subtitle,
+}: TabProps) {
+  const [focusProps, focusState] = useFocusState();
+
+  const isFocused = useMemo(
+    () => focusState === FocusState.FocusVisible,
+    [focusState]
+  );
+  const isFocusedWithin = useMemo(
+    () => focusState === FocusState.FocusWithinVisible,
+    [focusState]
+  );
+
+  const defaultActionProps = useDefaultAction(onSelect);
+
+  const [hoverProps, isHovered] = useHoverState();
+
+  const tabProps = mergeProps<HTMLDivElement>(
+    focusProps,
+    hoverProps,
+    defaultActionProps
+  );
+
+  const iconProps: Partial<React.ComponentProps<typeof Icon>> = useMemo(
+    () => ({
+      size: 'small',
+      role: 'presentation',
+      className: cx(tabIconStyles, {
+        [darkMode
+          ? tabIconSelectedDarkThemeStyles
+          : tabIconSelectedLightThemeStyles]: isSelected,
+        [tabIconFocusedStyles]: isFocused,
+      }),
+    }),
+    [darkMode, isFocused, isSelected]
+  );
+
+  return (
+    <div
+      className={cx(
+        tabStyles,
+        darkMode ? tabDarkThemeStyles : tabLightThemeStyles,
+        {
+          [selectedTabStyles]: isSelected,
+        }
+      )}
+      aria-selected={isSelected}
+      role="tab"
+      // Catch navigation on the active tab when a user tabs through Compass.
+      tabIndex={isSelected ? 0 : -1}
+      aria-controls={tabContentId}
+      title={`${subtitle} - ${title}`}
+      {...tabProps}
+    >
+      <div className={tabTitleContainerStyles}>
+        {renderIcon(iconProps)}
```

Comment:
If we want to be so much in control of the props that are passed to the icon, wouldn't it make sense to only allow to pass valid leafygreen glyph names here and nothing else? With the current implementation I can break the layout easily by setting my own size or a class name after spreading the props and the props are not generic enough to be spreadable to anything, like if I use another icon component that doesn't support size exactly as expected here it wouldn't work

Reply:
Sounds good, updated it to take a `iconGlyph` string with the `glyph` type from leafygreen.